vernac/lemmas.ml

```diff
@@ -441,15 +441,13 @@ let start_proof_com ?inference_hook kind thms hook =
     | Some decl ->
       Univdecls.interp_univ_decl_opt env0 (snd decl) in
   let evd, thms = List.fold_left_map (fun evd (sopt,(bl,t)) ->
-    let _evdref = ref evd in
-    let impls, ((env, ctx), imps) = interp_context_evars env0 _evdref bl in
-    let t', imps' = interp_type_evars_impls ~impls env _evdref t in
-    let evd = !_evdref in
+    let impls, ((env, ctx), evd, imps) = interp_context_evars env0 evd bl in
+    let evd, (t', imps') = interp_type_evars_impls ~impls env evd t in
     let flags = all_and_fail_flags in
     let flags = { flags with use_hook = inference_hook } in
     let evd = solve_remaining_evars flags env evd Evd.empty in
     let ids = List.map RelDecl.get_name ctx in
-    (* XXX: The nf_evar is critical !! *)
+    (* XXX: The nf_evar is critical !! XXX *)
```

Comment:
Why another XXX?

Reply:
To highlight how terrible is that we have to do that, or said otherwise, to highlight the types of the functions below are totally wrong.